base/tuple.jl

```diff
@@ -422,3 +422,104 @@ _tuple_any(f::Function, tf::Bool) = tf
 Returns an empty tuple, `()`.
 """
 empty(@nospecialize x::Tuple) = ()
+
+# optimizations for short tuples
+
+const ShortTuple = Union{
+    Tuple{},
+    Tuple{A} where {A},
+    Tuple{A, B} where {A, B},
+    Tuple{A, B, C} where {A, B, C},
+    Tuple{A, B, C, D} where {A, B, C, D},
+    Tuple{A, B, C, D, E} where {A, B, C, D, E},
+    Tuple{A, B, C, D, E, F} where {A, B, C, D, E, F},
+    Tuple{A, B, C, D, E, F, G} where {A, B, C, D, E, F, G},
+    Tuple{A, B, C, D, E, F, G, H} where {A, B, C, D, E, F, G, H},
+    Tuple{A, B, C, D, E, F, G, H, I} where {A, B, C, D, E, F, G, H, I},
+    Tuple{A, B, C, D, E, F, G, H, I, J} where {A, B, C, D, E, F, G, H, I, J},
+    Tuple{A, B, C, D, E, F, G, H, I, J, K} where {A, B, C, D, E, F, G, H, I, J, K},
+    Tuple{A, B, C, D, E, F, G, H, I, J, K, L} where {A, B, C, D, E, F, G, H, I, J, K, L},
+    Tuple{A, B, C, D, E, F, G, H, I, J, K, L, M} where {A, B, C, D, E, F, G, H, I, J, K, L, M},
+    Tuple{A, B, C, D, E, F, G, H, I, J, K, L, M, N} where {A, B, C, D, E, F, G, H, I, J, K, L, M, N},
+    Tuple{A, B, C, D, E, F, G, H, I, J, K, L, M, N, O} where {A, B, C, D, E, F, G, H, I, J, K, L, M, N, O},
+    Tuple{A, B, C, D, E, F, G, H, I, J, K, L, M, N, O, P} where {A, B, C, D, E, F, G, H, I, J, K, L, M, N, O, P}
```

Comment:
Note you can write `NTuple{16, Any}` instead of all of that... it's equivalent since tuple parameters are covariant